bundles/io/org.eclipse.smarthome.io.rest.sitemap/src/main/java/org/eclipse/smarthome/io/rest/sitemap/internal/PageChangeListener.java

```diff
@@ -201,7 +201,12 @@ public void stateUpdated(Item item, State state) {
                 event.item = EnrichedItemDTOMapper.map(item, false, null, null);
 
                 // adjust the state according to the widget type
-                event.item.state = itemUIRegistry.getState(w).toFullString();
+                String state = itemUIRegistry.getState(w).toFullString();
+                if (state.equals(event.item.state)) {
+                    event.state = "#item.state#";
```

Comment:
Static final string (or what other constants use). And maybe write add a comment why it was introduced

Reply:
Ok for the constant.
Maybe I will do this indirection only for big data like image.